Factory/HttpFoundationFactory.php

```diff
@@ -70,7 +74,6 @@ public function createRequest(ServerRequestInterface $psrRequest, bool $streamed
             $server,
             $streamed ? $psrRequest->getBody()->detach() : $psrRequest->getBody()->__toString()
         );
-        $request->headers->replace($psrRequest->getHeaders());
```

Comment:
How can the created request know about the headers of the psr one if we remove this line?

Reply:
They are already parsed few lines above. Without this line removed there is no way to use basic auth with Symfony - header is parsed during request object creation few lines above and then parsed user/password values are being overriden here.

Reply:
I kept the line and replaced "replace()" by "add()". This should provide better compat.